datafusion/core/src/physical_plan/udaf.rs

```diff
@@ -99,7 +100,7 @@ impl AggregateExpr for AggregateFunctionExpr {
     }
 
     fn create_accumulator(&self) -> Result<Box<dyn Accumulator>> {
-        (self.fun.accumulator)()
+        (self.fun.accumulator)(&self.data_type)
```

Comment:
The is the key change -- pass the return type when instantiating the accumulator